gui/default/fancyTreeGlobalDataWorker.js

```diff
@@ -0,0 +1,45 @@
+const generatedByTreeSelector = "// Generated by tree selector";
+
+var ignores = [];
+var invalidPattern = false;
+var inProgress = false;
+self.onmessage = function (msg) {
+	if (inProgress)
+		return;
+
+	inProgress = true;
+
+	var ignores = msg.data.ignorePatterns;
+	if (ignores[0] !== generatedByTreeSelector) {
+		invalidPattern = true;
+		ignores = [];
+	} else {
+		ignores = ignores.slice(1, -1)
+	}
+	self.postMessage({
+		data: formatGlobalTreeNodes(msg.data.data, ""),
+		invalidPattern: invalidPattern
+	});
+	inProgress = false;
+}
+
+function formatGlobalTreeNodes(data, parentKey, parentIgnored = true) {
+	var result = [];
+
+	data.forEach((node) => {
```

Comment:
The GUI currently doesn't use arrow functions, which makes it compatible with IE11 (and maybe WebView from Android 5, although there is conflicting information on the Web whether that WebView version supports arrow functions or not. However, Syncthing itself does officially support Android 5).

Unless absolutely necessary and used in multiple places, I'd suggest to use a regular function instead.